application/shared-kernel/ApiCore/SinglePageApp/SinglePageAppConfiguration.cs

```diff
@@ -80,16 +80,13 @@ public string GetHtmlTemplate()
     ///     It takes a few seconds from when the index.html is written to disk, until the JS and CSS are ready.
     /// </summary>
     [Conditional("DEBUG")]
-    private void AwaitSinglePageAppGeneration()
+    private void AwaitSinglePageAppGeneration(int maxRetries, TimeSpan retryInterval)
     {
-        var startNew = Stopwatch.StartNew();
-        while (startNew.Elapsed < TimeSpan.FromSeconds(10))
+        var retryCount = 0;
+        while (!File.Exists(_htmlTemplatePath) && retryCount < maxRetries)
         {
-            var spaCreationTimeUtc = new FileInfo(_htmlTemplatePath).CreationTimeUtc;
-            if (spaCreationTimeUtc > _startUpTime.AddSeconds(-10) && spaCreationTimeUtc < DateTime.UtcNow.AddSeconds(-2))

```

Comment:
We need this check to ensure that it’s not using the index.html from the previous build. We can update the while (startNew.Elapsed < TimeSpan.FromSeconds(10)) to a higher value. But 60 seconds is a long time, as there can be other reasons why it fails. Maybe 20 or 30 seconds?